shared/django_apps/timeseries/models.py

```diff
@@ -184,3 +185,94 @@ def is_backfilled(self) -> bool:
         if not self.created_at:
             return False
         return datetime.now() > self.created_at + timedelta(hours=1)
+
+
+class Testrun(ExportModelOperationsMixin("timeseries.testrun"), models.Model):
+    timestamp = models.DateTimeField(null=False, primary_key=True)
+
+    test_id = models.BinaryField(null=False)
+    flags_hash = models.BinaryField(null=True)
+
+    name = models.TextField(null=True)
+    classname = models.TextField(null=True)
+    testsuite = models.TextField(null=True)
+    computed_name = models.TextField(null=True)
+
+    outcome = models.TextField(null=False)
+
+    duration_seconds = models.FloatField(null=True)
+    failure_message = models.TextField(null=True)
+    framework = models.TextField(null=True)
+    filename = models.TextField(null=True)
+
+    repo_id = models.BigIntegerField(null=True)
+    commit_sha = models.TextField(null=True)
+    branch = models.TextField(null=True)
+
+    flags = ArrayField(models.TextField(), null=True)
+    upload_id = models.BigIntegerField(null=True)
+
+    class Meta:
+        app_label = TIMESERIES_APP_LABEL
+        indexes = [
+            models.Index(
+                fields=["repo_id", "test_id", "flags_hash"],
+            ),
+        ]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["repo_id", "test_id", "flags_hash"],
+                name="flags_hash_test_id_unique",
+            ),
```

Comment:
I believe a unique constraint already implicitly creates an index that can be used for queries. which means this would be duplicated.

on the other hand, I don’t see any indices being defined on the materialized views. Do we not need those?